R/paginate_listing.R

```diff
@@ -76,45 +73,25 @@ paginate_listing <- function(lsting,
     colwidths = colwidths,
     tf_wrap = tf_wrap,
     max_width = max_width,
-    rep_cols = length(get_keycols(lsting)),
+    rep_cols = rep_cols,
     verbose = verbose
   )
 
-  vert_pags <- lapply(
-    indx$pag_row_indices,
-    function(ii) lsting[ii, ]
-  )
-  dispnames <- listing_dispcols(lsting)
-  full_pag <- lapply(
-    vert_pags,
-    function(onepag) {
-      if (!is.null(indx$pag_col_indices)) {
-        lapply(
-          indx$pag_col_indices,
-          function(jj) {
-            res <- onepag[, dispnames[jj], drop = FALSE]
-            listing_dispcols(res) <- intersect(dispnames, names(res))
-            res
-          }
-        )
-      } else {
-        list(onepag)
-      }
-    }
-  )
-
-  ret <- unlist(full_pag, recursive = FALSE)
-  ret
+  if (printable_form) {
+    lapply(pages, toString, widths = NULL, tf_wrap = tf_wrap, max_width = max_width)
+  } else {
+    pages
+  }
 }
```

Comment:
I did this only to have it in a printable form (but anyhow did go for something like it in the tests, I could take this out and keep the testing functions